### PR TITLE
Configure import grouping and granularity in rustfmt

### PR DESCRIPTION
**Pull request by SimonSapin**
Per discussion in https://github.com/apollographql/apollo-rs/pull/868#discussion_r1651705217 except I picked `group_imports = One` as I slightly prefer it over `StdExternalCrate`. I don’t mind much which style we pick as long as we pick one.

### CODE REVIEW

**Comment by SimonSapin** [comment]
To avoid merge conflicts the easiest is likely:

1. Merge https://github.com/apollographql/apollo-rs/pull/868 into `main` when it’s ready
2. Revert the *Run `cargo +nightly fmt --all` with the new config* commit in this branch. (It’s included here to show what the result looks like)
3. Merge `main` into this branch
4. Run `cargo +nightly fmt --all` on the merge to make another version of that commit
5. Push to update this PR

**Review by lrlna — APPROVED** [review]
happy to do what we do in the router!

File: crates/apollo-compiler/src/validation/union_.rs [hunk 1]
```diff
@@ -1,7 +1,10 @@
-use crate::schema::{self, ExtendedType, UnionType};
+use crate::ast;
+use crate::schema::ExtendedType;
+use crate::schema::UnionType;
+use crate::schema::{self};
```
Comment (goto-bus-stop) [inline crates/apollo-compiler/src/validation/union_.rs:4]:
```suggestion
use crate::schema;
```

File: crates/apollo-compiler/examples/file_watcher.rs [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
 use apollo_compiler::Schema;
-use notify::{Config, EventKind, PollWatcher, RecursiveMode, Watcher};
+use notify::Config;
+use notify::EventKind;
+use notify::PollWatcher;
+use notify::RecursiveMode;
+use notify::Watcher;
+use std::collections::HashMap;
+use std::fs::DirEntry;
+use std::fs::{self};
```
Comment (goto-bus-stop) [inline crates/apollo-compiler/examples/file_watcher.rs:13]:
```suggestion
use std::fs;
```

**Review by goto-bus-stop — COMMENTED** [review]
We have a `cargo xtask lint` that should also pass the `+nightly` flag.

File: .circleci/config.yml [hunk 3]
```diff
@@ -100,8 +100,13 @@ jobs:
           name: Run cargo clippy
           command: cargo clippy --all-targets --all-features -- -D warnings
       - run:
-          name: Run cargo fmt check
-          command: rustup component add rustfmt && cargo fmt --all -- --check
+          name: Install rustfmt nightly
+          command: >
+            rustup install nightly --profile minimal &&
+            rustup component add --toolchain nigthly rustfmt
```
Comment (goto-bus-stop) [inline .circleci/config.yml:106]:
```suggestion
            rustup component add --toolchain nightly rustfmt
```

**Pull request merged**